photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala

```diff
@@ -186,9 +186,12 @@ final class Driver(val sc: SparkContext, val params: GameParams, implicit val lo
 
     data.map { dataframe =>
       if (params.checkData) {
-        val numBad = dataframe
-          .map { row => if (row.getAs[Double](weightColumnName) <= 0.0) 1 else 0 }
-          .reduce(_ + _)
+        val numBad = dataframe.filter(s"$weightColumnName <= 0.0").count()
```

Comment:
Consider using expressing the predicate in Scala rather than sticking it all in a String:

```scala
val numBad = dataframe.filter(col("weightColumnName") <= 0.0).count()
```